File: python intermediario/Lista de Tarefas.py
def exibir_tarefas(tarefas):
    if not tarefas:
        print("Nenhuma tarefa encontrada!")
    else:
        print("\nLista de Tarefas:")
        for id, (tarefa, concluida) in tarefas.items():
            status = "Concluída" if concluida else "Pendente"
            print(f"{id}. {tarefa} - {status}")

def adicionar_tarefa(tarefas):
    tarefa = input("Digite a tarefa: ")
    id_tarefa = len(tarefas) + 1  
    tarefas[id_tarefa] = (tarefa, False)  
    print(f"Tarefa '{tarefa}' adicionada com sucesso!")

def marcar_concluida(tarefas):
    try:
        id_tarefa = int(input("Digite o ID da tarefa a ser marcada como concluída: "))
        if id_tarefa in tarefas:
            tarefa, _ = tarefas[id_tarefa]
            tarefas[id_tarefa] = (tarefa, True)  
            print(f"Tarefa '{tarefa}' marcada como concluída!")
        else:
            print("ID de tarefa inválido!")
    except ValueError:
        print("Por favor, insira um número válido para o ID da tarefa.")

def excluir_tarefa(tarefas):
    try:
        id_tarefa = int(input("Digite o ID da tarefa a ser excluída: "))
        if id_tarefa in tarefas:
            tarefa, _ = tarefas.pop(id_tarefa)  
            print(f"Tarefa '{tarefa}' excluída com sucesso!")
        else:
            print("ID de tarefa inválido!")
    except ValueError:
        print("Por favor, insira um número válido para o ID da tarefa.")

def menu():
    tarefas = {}
    
    while True:
        print("\nEscolha uma opção:")
        print("1. Adicionar tarefa")
        print("2. Marcar tarefa como concluída")
        print("3. Excluir tarefa")
        print("4. Exibir todas as tarefas")
        print("5. Sair")
        
        opcao = input("Digite o número da opção: ")
        
        if opcao == "1":
            adicionar_tarefa(tarefas)
        elif opcao == "2":
            marcar_concluida(tarefas)
        elif opcao == "3":
            excluir_tarefa(tarefas)
        elif opcao == "4":
            exibir_tarefas(tarefas)
        elif opcao == "5":
            print("Saindo... Até logo!")
            break
        else:
            print("Opção inválida, por favor, tente novamente!")

if __name__ == "__main__":
    menu()

File: exemplo java/lista de tarefas.java
import java.util.ArrayList;
import java.util.Scanner;

public class ListaDeTarefas {

    static class Tarefa {
        String descricao;
        boolean concluida;

        public Tarefa(String descricao) {
            this.descricao = descricao;
            this.concluida = false;
        }

        @Override
        public String toString() {
            return (concluida ? "[Concluída] " : "[Pendente] ") + descricao;
        }

        public void concluir() {
            concluida = true;
        }

        public void desmarcar() {
            concluida = false;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        ArrayList<Tarefa> tarefas = new ArrayList<>();

        while (true) {
            System.out.println("\nLista de Tarefas:");
            System.out.println("1. Adicionar tarefa");
            System.out.println("2. Listar tarefas");
            System.out.println("3. Marcar tarefa como concluída");
            System.out.println("4. Remover tarefa");
            System.out.println("5. Sair");
            System.out.print("Escolha uma opção: ");
            int opcao = scanner.nextInt();
            scanner.nextLine(); 

            switch (opcao) {
                case 1:
                    System.out.print("Digite a descrição da tarefa: ");
                    String descricao = scanner.nextLine();
                    tarefas.add(new Tarefa(descricao));
                    System.out.println("Tarefa adicionada com sucesso!");
                    break;

                case 2:
                    System.out.println("\nLista de Tarefas:");
                    for (int i = 0; i < tarefas.size(); i++) {
                        System.out.println((i + 1) + ". " + tarefas.get(i));
                    }
                    break;

                case 3:
                    System.out.print("Digite o número da tarefa para marcar como concluída: ");
                    int concluirIndex = scanner.nextInt() - 1;
                    if (concluirIndex >= 0 && concluirIndex < tarefas.size()) {
                        tarefas.get(concluirIndex).concluir();
                        System.out.println("Tarefa marcada como concluída!");
                    } else {
                        System.out.println("Número de tarefa inválido.");
                    }
                    break;

                case 4:
                    System.out.print("Digite o número da tarefa para remover: ");
                    int removerIndex = scanner.nextInt() - 1;
                    if (removerIndex >= 0 && removerIndex < tarefas.size()) {
                        tarefas.remove(removerIndex);
                        System.out.println("Tarefa removida com sucesso!");
                    } else {
                        System.out.println("Número de tarefa inválido.");
                    }
                    break;

                case 5:
                    System.out.println("Saindo...");
                    scanner.close();
                    return;

                default:
                    System.out.println("Opção inválida.");
            }
        }
    }
}
